core/shared/src/main/scala/scalaz/zio/Exit.scala

```diff
@@ -282,23 +285,43 @@ object Exit extends Serializable {
         }
         .reverse
 
+    final def traces: List[ZTrace] =
+      self
+        .fold(List.empty[ZTrace]) {
+          case (z, f @ Fail(_))     => f.trace.fold(z)(_ :: z)
+          case (z, d @ Die(_))      => d.trace.fold(z)(_ :: z)
+          case (z, i @ Interrupt()) => i.trace.fold(z)(_ :: z)
+          case (z, r @ Refail(_))   => r.trace :: z
+        }
+        .reverse
+
     final def fold[Z](z: Z)(f: PartialFunction[(Z, Cause[E]), Z]): Z =
       (f.lift(z -> self).getOrElse(z), self) match {
         case (z, Then(left, right)) => right.fold(left.fold(z)(f))(f)
         case (z, Both(left, right)) => right.fold(left.fold(z)(f))(f)
+        case (z, Refail(cause))     => cause.fold(z)(f)
 
         case (z, _) => z
       }
 
+    /**
+     * Retrieve the first checked error on the `Left` if available,
+     * if there are no checked errors return the rest of the `Cause`
+     * that is known to contain only `Die` or `Interrupt` causes.
+     * */
     final def failureOrCause: Either[E, Cause[Nothing]] = self.failures.headOption match {
       case Some(error) => Left(error)
       case None        => Right(self.asInstanceOf[Cause[Nothing]]) // no E inside this cause, can safely cast
     }
 
+    /**
+     * Remove all `Fail` and `Interrupt` nodes from this `Cause`,
+     * return only [[Die]] cause/finalizer defects.
```

Comment:
Should be `` `Die` `` otherwise Scaladoc complains.

Reply:
done!